deploy-deps.sh

```diff
@@ -14,26 +14,63 @@ main() {
 }
 
 check_req(){
-    # declare the requirements
-    local requirements=(kubectl openssl)
+    # Helper function to compare versions
+    # Returns 0 if installed_version >= min_required_version, 1 otherwise
+    version_ge() {
+        local installed_version="$1"
+        local min_required_version="$2"
+        # Use sort -V (version sort) to compare versions
+        printf '%s\n%s\n' "$min_required_version" "$installed_version" | sort -V -C
```

Comment:
I'm not sure how robust this is. It depends on the local implementation of `sort` the user has, and it might not support those flags (AI says sort -V is a non-portable GNU extension and will fail on systems like macOS). It also depends on the structure of the version strings, which I'm not sure we want to depend on.

I wonder if we, at the very least, would want to provide a mechanism to skip this check based on some input.

Reply:
> What other way do you suggest to get the versions?

It's more about parsing than getting. Suppose that there's some sort of special build the user is using which is new enough but contains a weird suffix.

> We can use other method than sort -V, maybe a strict bash flow

Need to make sure this does become a maintenance burden. AI suggests `awk`, but personally I think this becomes "write-only code" very easily.

Maybe we can print the failure (either for wrong version or command failures) and prompt the user to either fix the installed version or run with a flag that will skip the check.

Reply:
BTW, the CI is failing because the `ubuntu-latest` which we are using does not contain the minimum required openssl version.
In the requirements it is mentioned that the openSSL version should be v3.2.2 or higher, while we have v3.0.13.
So we probably need either to lower the requirement (if possible) or to somehow install openssl with the appropriate version.